MapboxVisionTests/VisionManagerTests.swift

```diff
@@ -0,0 +1,137 @@
+@testable import MapboxVision
+import XCTest
+
+class VisionManagerTests: XCTestCase {
+    var visionManager: VisionManager?
+    var dependencies: VisionDependencies?
+
+    override func setUp() {
+        super.setUp()
+
+        dependencies = VisionDependencies(
+            native: MockNative(),
+            synchronizer: MockSynchronizable(),
+            recorder: MockSessionRecorder(),
+            dataProvider: MockDataProvider(),
+            deviceInfo: DeviceInfoProvider()
+        )
+
+        self.visionManager = VisionManager(dependencies: dependencies!, videoSource: MockVideoSource())
+    }
+
+    func testNoCrashOnVisionManagerDeallocWithoutDestroy() {
```

Comment:
Do we test there's no crash? A bit confusing as I see only checks there's no exception. Here and below.

Reply:
yes, but we can't write explicit test for EXC_BAD_ACCESS